packages/mui-base/src/MenuUnstyled/useMenu.ts

```diff
@@ -71,7 +71,10 @@ export default function useMenu(parameters: UseMenuParameters) {
   } = useListbox({
     options: Object.keys(menuItems),
     optionStringifier: (id: string) =>
-      menuItems[id].ref.current?.innerText || menuItems[id].ref.current?.innerHTML || id,
+      menuItems[id].label ||
+      menuItems[id].ref.current?.innerText ||
+      menuItems[id].ref.current?.innerHTML ||
+      id,
```

Comment:
The `id` is an internal identifier and it doesn't mean anything to the user. I wouldn't use it in this context. If there's no content in the element, let's not make it navigatable to.

Also, innerHTML doesn't feel right as likely it'll start with a `<` if innerText is undefined. innerText should be enough.

Reply:
> The id is an internal identifier and it doesn't mean anything to the user. I wouldn't use it in this context. If there's no content in the element, let's not make it navigable to.

Okay let me do that.

In this case, I will make similar change to `useSelect` (and `useListbox`) to make option not navigable if it can't be stringified ([yes this is a word apparently](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)).

> Also, innerHTML doesn't feel right as likely it'll start with a < if innerText is undefined. innerText should be enough.

The reason I added `innerHTML` after `innerText` is because when running [tests](https://github.com/ivan-ngchakming/material-ui/blob/5c6e5e14c1bd5fcb86867e039730fcc1b8680be3/packages/mui-base/src/MenuUnstyled/MenuUnstyled.test.tsx#L114-L140) on `MenuUnstyled`, `innerText` is undefined, breaking the keyboard text navigation, I fixed it by adding `innerHTML` as fall back.

Reply:
The tests fail because `innerText` is not supported in JSDOM (see https://github.com/jsdom/jsdom/issues/1245). You can skip running this particular test with JSDOM and only run it on real browsers (using `yarn test:karma`) by adding the following code to the top of the test case:

```ts
if (/jsdom/.test(window.navigator.userAgent)) {
  this.skip();
}
```

Note that `this` refers to the test function, so you'll have to change it from an arrow to an anonymous function for it to work.